framework/test/mocha/unit/modules/chain/submodules/process_transactions.js

```diff
@@ -41,16 +80,435 @@ describe('ProcessTransactions', () => {
 		},
 	};
 
-	beforeEach(async () => {
+	beforeEach(done => {
 		// Act
-		processTransactions = new ProcessTransactions(() => {}, {
-			components: {},
-			logic: {},
+		processTransactions = new ProcessTransactions(() => {
+			processTransactions.onBind(paramScope);
+
+			// Add matcher to transactions
+			trs1.matcher = () => true;
+			trs2.matcher = () => true;
+
+			// Add prepare steps to transactions
+			trs1.prepare = sinonSandbox.stub();
+			trs2.prepare = sinonSandbox.stub();
+
+			// Add apply steps to transactions
+			trs1.apply = sinonSandbox.stub();
+			trs2.apply = sinonSandbox.stub();
+
+			// Add undo steps to transactions
+			trs1.undo = sinonSandbox.stub();
+			trs2.undo = sinonSandbox.stub();
+
+			ProcessTransactions.__set__('StateStore', StateStoreStub);
+			ProcessTransactions.__set__('roundInformation', roundInformationMock);
+
+			done();
+		}, paramScope);
+	});
+
+	afterEach(async () => {
+		sinonSandbox.restore();
+		sinonSandbox.reset();
+
+		roundInformationMock.apply.reset();
+		roundInformationMock.undo.reset();
+
+		StateStoreStub.resetHistory();
+		stateStoreMock.restoreSnapshot.reset();
+		stateStoreMock.createSnapshot.reset();
+		stateStoreMock.account.restoreSnapshot.reset();
+		stateStoreMock.account.createSnapshot.reset();
+		stateStoreMock.transaction.add.reset();
```

Comment:
`sinonSandbox.reset` should be covering all these reset calls. Are they really needed?

Reply:
The call to `sinonSandbox.reset` is not resetting all stubs, don't exactly debugged why. Will look at it later.